### ISSUE TITLE
Improve support for localization

### ISSUE THREAD

**Issue by ndw** [docbook/xslTNG#132]
At present, it appears you need a customization layer to change a variable. That's madness, it should be a parameter, surely?

**Comment by wirepatch** [comment]
Just fetched stock https://github.com/docbook/xslTNG/releases/download/1.7.1/docbook-xslTNG-1.7.1.zip stylesheets trying German localization using the supplied Python script after retrieving dependency jars using Maven:

```
$ ./bin/docbook test.xml -xsl:xslt/docbook.xsl  default-language=de -o:output/index.html 
No localization data for de
No localization data for de
No localization data for de
No localization data for de
No localization data for de
No localization data for de
```
The sample is a minimal doc:

```
<book version="5.1" xmlns="http://docbook.org/ns/docbook"
      xmlns:xlink="http://www.w3.org/1999/xlink"
     ... NS decls omitted for brevity ... >
  <info>
    <title>Erster Test</title>
  </info>
  <chapter xml:id="chap1">
    <title>Test 1</title>
    <para>Kapitel 1.</para>
  </chapter>
</book>
```
Adding `<book version="5.1"`***`xml:lang="de`***`...` yields the very same result. File 'xslt/locale/de.xml` does contain adequate definitions but is not being considered with respect to the generated output.

Hunted the docs and the parameter reference in particular without finding a clue.  This always worked for the Docbook Xslt »legacy« stylesheets.

**Comment by wirepatch** [comment]
... and creating a customization layer defining `<xsl:variable name="default-language" select="'de'"/>` doesn't work either. Only `... select="'en'" ...` processes without warnings.

**Comment by ndw** [comment]
I managed to mangle the PR for this fix, so it's part of #164 instead of being it's own PR. I'm not quite sure how I did that :-(.

With respect to localization, I think I had an idea that the languages should be resolve statically and it was pretty clear I never actually finished the design. On reflection, I decided to simplify things. The new design is documented in the guide, but in brief:

1. The `$default-language` parameter identifies the default language
2. The `$additional-languages` parameter identifies additional localizations that you want to make available.

By default, the `$additional-languages` parameter is the empty sequence (`()`) which has the special meaning that the stylesheets will look for `//@xml:lang` attribute values and determine the localizations automatically.

**Comment by wirepatch** [comment]
This solution partly works but fails for modular documents. The following monolithic sample transforms as expected:
```
<book version="5.1" xmlns="http://docbook.org/ns/docbook">
  <info>
    <title>Multilingual test</title>
  </info>

  <chapter>
    <title>English text</title>
    <para>Life goes on ...</para>
  </chapter>

  <chapter xml:lang="de">
    <title>Deutscher Text</title>
    <para>Das Leben geht weiter</para>
  </chapter>
</book>
```
However moving the second chapter to a module `germanChapter.xml`:
```
<chapter version="5.1" xmlns="http://docbook.org/ns/docbook"
	 xml:lang="de">
    <title>Deutscher Text</title>

    <para>Das Leben geht weiter</para>
</chapter>
```
... and using a master document: 
```
<book version="5.1" xmlns="http://docbook.org/ns/docbook"
      xmlns:xi="http://www.w3.org/2001/XInclude">
  <info>
    <title>Multilingual Xinclude test</title>
  </info>
  
  <chapter>
    <title>English text</title>
    <para>Life goes on ...</para>
  </chapter>

  <xi:include href="germanChapter.xml" xpointer="element(/1)"/>

</book>
```
... fails:
```
.../bin/docbook germanXinclude.xml  -xsl:/.../docbook.xsl 
No localization data for de 
...
```
